### PR TITLE
genie: Conan v2 support and add version 1170

### PR DESCRIPTION
**Pull request by RazielXYZ**
Specify tool name and version:  **genie/1170**

This is a build tool, required for a few packages that I want to add, like bx, bimg and bgfx.

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/adding_packages/README.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/genie/all/conandata.yml [hunk 1]
```diff
@@ -8,3 +8,6 @@ sources:
   "1160":
     url: "https://github.com/bkaradzic/GENie/archive/f9bd455a8439dbcb807816c0be9e4aedf5991bc3.zip"
     sha256: "6fe7e2f0107c965ebe9d099b0168128dfcaee67d7368d4b15b42d80808e3d373"
+  "1170":
```
Comment (uilianries) [inline recipes/genie/all/conandata.yml:11]: https://github.com/bkaradzic/GENie/commit/81e594dc974866b5d301711258c774cbf3c68883#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R25

Okay

File: recipes/genie/all/conanfile.py [hunk 2]
```diff
@@ -14,36 +19,40 @@ class GenieConan(ConanFile):
...
+
+    def validate(self):
+        if hasattr(self, "settings_build") and cross_building(self):
+            raise ConanInvalidConfiguration("Cross building is not yet supported. Contributions are welcome")
+
     def build_requirements(self):
-        if self.settings.compiler == "Visual Studio":
-            self.build_requires("cccl/1.1")
+        if is_msvc(self):
+            self.build_requires("cccl/1.3")
```
Comment (uilianries) [inline recipes/genie/all/conanfile.py:41]:
```suggestion
            self.tool_requires("cccl/1.3")
```
https://docs.conan.io/en/latest/devtools/build_requires.html#build-and-host-contexts

File: recipes/genie/all/conanfile.py [hunk 3]
```diff
@@ -14,36 +19,40 @@ class GenieConan(ConanFile):
...
 
         if self.settings.os == "Windows" and self._settings_build.os == "Windows":
             if "make" not in os.environ.get("CONAN_MAKE_PROGRAM", ""):
                 self.build_requires("make/4.3")
 
-            if not tools.get_env("CONAN_BASH_PATH"):
+            if self._settings_build.os == "Windows" and not self.conf.get("tools.microsoft.bash:path", default=False, check_type=bool):
```
Comment (uilianries) [inline recipes/genie/all/conanfile.py:47]:
```suggestion
            if self._settings_build.os == "Windows" and not self.conf.get("tools.microsoft.bash:path", default=False, check_type=str):
```
It's a path

**Review by uilianries — COMMENTED** [review]
Thank you for your contribution. Please, update build_requirements according to the autotools template: https://github.com/conan-io/conan-center-index/blob/master/docs/package_templates/autotools_package/all/conanfile.py#L84

File: recipes/genie/all/test_v1_package/scripts/genie.lua [hunk 4]
```diff
@@ -0,0 +1,23 @@
+newoption {
```
Comment (uilianries) [inline recipes/genie/all/test_v1_package/scripts/genie.lua:1]: please, re-use from test package with relative path and remove this copied file.

File: recipes/genie/all/test_v1_package/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,10 @@
+from conans import ConanFile, tools
+
+
+class TestPackageConan(ConanFile):
+
+    settings = "os", "arch"
+
+    def test(self):
+        if not tools.cross_building(self):
+            self.run('genie ninja --scripts="{}"'.format(self.source_folder), run_environment=True)
```
Comment (uilianries) [inline recipes/genie/all/test_v1_package/conanfile.py:10]:
```suggestion
            self.run('genie ninja --scripts="{}"'.format(os.path.join(self.source_folder, "..", "test_package")), run_environment=True)
```

**Comment by RazielXYZ** [comment]
The only way I could actually get it to generate projects that way was by using cwd rather than the scripts argument, hopefully that's okay.

**Review by prince-chrismc — APPROVED** [review]

**Review by uilianries — APPROVED** [review]
LGTM

**Comment by prince-chrismc** [comment]
The recover attempt failed and did not post anything 😕

**Pull request merged**